doc/development/adding_operators.rst

```diff
@@ -0,0 +1,356 @@
+            shape = qml.math.shape(angle)
+            if len(shape) > 1:
+                raise ValueError(f"Expected a scalar angle; got angle of shape {shape}.")
+
+            #------------------------------------
+
+            # do_flip is not trainable but influences the action of the operator,
+            # which is why we define it to be a hyperparameter
+            self._hyperparameters = {
+                "do_flip": do_flip
+            }
+
+            # we extract all wires that the operator acts on,
+            # relying on the Wire class arithmetic
+            all_wires = qml.wires.Wires(wire_rot) + qml.wires.Wires(wire_flip)
+
+            # The parent class expects all trainable parameters to be fed as positional
+            # arguments, and all wires acted on fed as a keyword argument.
+            # The id keyword argument allows users to give their instance a custom name.
+            # The do_queue keyword argument specifies whether or not
+            # the operator is queued when created in a tape context.
+            super().__init__(angle, wires=all_wires, do_queue=do_queue, id=id)
+
+        @property
+        def num_params(self):
+            # if it is known before creation, define the number of parameters to expect here,
+            # which makes sure an error is raised if the wrong number was passed
+            return 1
+
+        @staticmethod
+        def compute_decomposition(angle, wires, do_flip):  # pylint: disable=arguments-differ
+            # Overwriting this method defines the decomposition of the new gate, as it is
+            # called by Operator.decomposition().
+            # The general signature of this function is (*parameters, wires, **hyperparameters).
+            op_list = []
+            if do_flip:
+                op_list.append(qml.PauliX(wires=wires[1]))
+            op_list.append(qml.RX(angle, wires=wires[0]))
+            return op_list
+
+        def adjoint(self):
+            # the adjoint operator of this gate simply negates the angle
+            return FlipAndRotate(-self.parameters[0], self.wires[0], self.wires[1], do_flip=self.hyperparameters["do_flip"])
+
+The new gate can now be created as follows:
+
+.. code-block:: python
+
+    >>> op = FlipAndRotate(0.1, wire_rot="q3", wire_flip="q1", do_flip=True)
+    >>> op
+    FlipAndRotate(0.1, wires=['q3', 'q1'])
+    >>> op.decomposition()
+    [PauliX(wires=['q1']), RX(0.1, wires=['q3'])]
+    >>> op.adjoint()
+    FlipAndRotate(-0.1, wires=['q3', 'q1'])
+
+The new gate can be used in devices, which access the decomposition to interpret it:
+
+.. code-block:: python
+
+    from pennylane import numpy as np
+
+    dev = qml.device("default.qubit", wires=["q1", "q2", "q3"])
+
+    @qml.qnode(dev)
+    def circuit(angle):
+        FlipAndRotate(angle, wire_rot="q1", wire_flip="q1")
+        return qml.expval(qml.PauliZ("q1"))
+
+>>> a = np.array(3.14)
+>>> circuit(a)
+-0.9999987318946099
```

Comment:
Actually, looking at this example again, what is the reasoning behind subclassing an operator vs. simply the function definition in the comment above? 🤔 

In what cases does subclassing provide more power/flexibility than simply defining a function?

Reply:
I worry users might find this page in the docs, and attempt to subclass, rather than simply writing
```python
def FlipAndRotate(...):
```

Reply:
Well, it is a dev section, and this is of course just an example...subclassing opens up a lot more possibilities, such as defining a name and id, a shape,...

I will add a note about this!